src/Markdig/Extensions/MediaLinks/HostProviderBuilder.cs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Alexandre Mutel. All rights reserved.
+// This file is licensed under the BSD-Clause 2 license.
+// See the license.txt file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Markdig.Extensions.MediaLinks
+{
+    public class HostProviderBuilder
+    {
+        private sealed class DelegateProvider : IHostProvider
+        {
+            public string HostPrefix { get; set; }
+            public Func<Uri, string> Delegate { get; set; }
+            public bool AllowFullScreen { get; set; } = true;
+            public string Class { get; set; }
+
+            public bool TryHandle(Uri mediaUri, bool isSchemaRelative, out string iframeUrl)
+            {
+                if (!mediaUri.Host.StartsWith(HostPrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    iframeUrl = null;
+                    return false;
+                }
+                iframeUrl = Delegate(mediaUri);
+                return !string.IsNullOrEmpty(iframeUrl);
+            }
+        }
+
+        /// <summary>
+        /// Create a <see cref="IHostProvider"/> with delegate handler.
+        /// </summary>
+        /// <param name="hostPrefix">Prefix of host that can be handled.</param>
+        /// <param name="handler">Handler that generate iframe url, if uri cannot be handled, it can return <see langword="null"/>.</param>
+        /// <param name="allowFullScreen">Should the generated iframe has allowfullscreen attribute.</param>
+        /// <param name="iframeClass">"class" attribute of generated iframe.</param>
+        /// <returns>A <see cref="IHostProvider"/> with delegate handler.</returns>
+        public static IHostProvider Create(string hostPrefix, Func<Uri, string> handler, bool allowFullScreen = true, string iframeClass = null)
+        {
+            if (string.IsNullOrEmpty(hostPrefix))
+                throw new ArgumentException("hostPrefix is null or empty.", nameof(hostPrefix));
+            if (handler == null)
+                throw new ArgumentNullException(nameof(handler));
+            return new DelegateProvider { HostPrefix = hostPrefix, Delegate = handler, AllowFullScreen = allowFullScreen, Class = iframeClass };
+        }
+
+        public static Dictionary<string, IHostProvider> KnownHosts { get; }
```

Comment:
Could you make this property internal? 

```suggestion
        internal static Dictionary<string, IHostProvider> KnownHosts { get; }
```